### PR TITLE
Created Hey Function test file

### PR DESCRIPTION
**Pull request by coolwednesday**

### CODE REVIEW

File: shape_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func test_GetDetails(t *testing.T) {
```
Comment (aryanmehrotra) [inline shape_test.go:8]: Test function names should start with Test_

File: shape_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	tests := []struct {
+		name   string
+		input  shape
+		output string
+	}{
+		{name: "Rectangle", input: rectangle{length: 10, breadth: 8}, output: "Shape Details => Area: 80.0 Perimeter: 36.0"},
+		{name: "Circle", input: circle{radius: 10}, output: "Shape Details => Area: 314.0 Perimeter: 62.8"},
+		{name: "Triangle", input: triangle{base: 4, height: 3}, output: "Shape Details => Area:6.0  Perimeter: 12.0"},
+		{name: "Square", input: square{length: 10}, output: "Shape Details => Area: 100.0 Perimeter: 40.0"},
```
Comment (aryanmehrotra) [inline shape_test.go:17]: you should also write tests to cover edge cases

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]
